src/javascript/node-package.ts

```diff
@@ -1552,6 +1621,33 @@ export enum NpmAccess {
   RESTRICTED = "restricted",
 }
 
+/**
+ * Configure Yarn Berry
+ */
+export interface YarnBerryOptions {
+  /**
+   * A fully specified version to use for yarn (e.g., x.x.x)
+   *
+   * @default - 4.0.1
+   */
+  readonly version?: string;
+
+  /**
+   * The yarnrc configuration.
+   *
+   * @default - a blank Yarn RC file
+   */
+  readonly yarnRcOptions?: YarnrcOptionsV3 | YarnrcOptionsV4;
```

Comment:
```suggestion
  readonly yarnRcOptions?: YarnrcOptions;
```
This should be an enum like class:

```ts
YarnrcOptions.v3(options: YarnrcOptionsV3);
YarnrcOptions.v4(options: YarnrcOptionsV4);
```

Alternatively we can forgo the strict typing here, have a single `YarnrcOptions` interface and validate differently in the `v3` / `v4` static methods.

Reply:
Or maybe it's even just the same interface and we validate it based on the version that is set. Differences between v3/v4 can be documented.

Reply:
I went ahead and used the same interface. I marked removed options with a `@deprecated` JSDoc annotation and added validation logic to warn users about invalid config usage.